src/Export/Skills/act_str.txt

```diff
@@ -632,10 +632,79 @@ local skills, mod, flag, skill = ...
 
 #skill FrozenLegion
 #flags spell attack area melee
+statMap = {
+		["frozen_legion_%_chance_to_summon_additional_statue"] = {
+			mod("FrozenLegionExtraStatueChance", "BASE", nil)
+		}
+	},
+#baseMod mod("FrozenLegionMaxStatues", "BASE", 6)
 #mods
 
 #skill FrozenSweep
 #flags spell attack area melee
+	parts = {
+		{
+			name = "1 charge",
+		},
+		{
+			name = "Wait for max charges",
+		},
+	},
+	preDamageFunc = function(activeSkill, output, breakdown)
+		local t_insert = table.insert
+		local s_format = string.format
+		local parentSkill
+		local m_ceil = math.ceil
+		local m_max = math.max
+		activeSkill.skillData.showAverage = false
+		activeSkill.skillFlags.showAverage = false
+		activeSkill.skillFlags.notAverage = true	
+		
+		for _, skill in ipairs(activeSkill.actor.activeSkillList) do
+			if skill.activeEffect.grantedEffect.name == "Frozen Legion" and activeSkill.actor.mainSkill.socketGroup.slot == activeSkill.socketGroup.slot then
+				parentSkill = skill
+				break
+			end
+		end
+
+		if parentSkill.skillModList:Flag(parentSkill.skillCfg, "DisableSkill") and not parentSkill.skillModList:Flag(parentSkill.skillCfg, "EnableSkill") then
+			return
+		end
+		
+		activeSkill.activeEffect.grantedEffect.castTime = 0
+		parentSkill.activeEffect.grantedEffect.castTime = 0
+		
+		activeSkill.skillData.cooldown = parentSkill.skillData.cooldown
+		local cooldownOverride = parentSkill.skillModList:Override(parentSkill.skillCfg, "CooldownRecovery")
+		local cooldown = cooldownOverride or (parentSkill.skillData.cooldown + parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "CooldownRecovery")) / m_max(0, calcLib.mod(parentSkill.skillModList, parentSkill.skillCfg, "CooldownRecovery"))
+		output.Cooldown = m_ceil(cooldown * data.misc.ServerTickRate) / data.misc.ServerTickRate
+		activeSkill.skillData.hitTimeOverride = output.Cooldown
+
+		local ExtraHitChance = parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "FrozenLegionExtraStatueChance") or 0
+		local MaxStatues = parentSkill.skillModList:Sum("BASE", parentSkill.skillCfg, "FrozenLegionMaxStatues") or 1
+		
+		local statuesWaitedFor
+		local dpsMultiplier = ExtraHitChance
+
+		if activeSkill.skillPart == 1 then
+			statuesWaitedFor = 1
+		elseif activeSkill.skillPart == 2 then
+			statuesWaitedFor = MaxStatues
+		end
+		activeSkill.skillData.averageBurstHits = statuesWaitedFor + (ExtraHitChance > 0 and ExtraHitChance / 100 or 0)
```

Comment:
```suggestion
		activeSkill.skillData.averageBurstHits = statuesWaitedFor + ExtraHitChance / 100
```
You have ensured this is always a number I don't see the reason to bother check it is greater than zero either.